src/PrestaShopBundle/Kernel/ModuleRepositoryFactory.php

```diff
@@ -0,0 +1,155 @@
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShopBundle\Kernel;
+
+use Doctrine\DBAL\DriverManager;
+use PrestaShop\PrestaShop\Adapter\Environment;
+
+/**
+ * Class ModuleRepositoryFactory is used to build the ModuleRepository in context where symfony container is not
+ * available or not yet initialised (ex: AppKernel, PrestaShop\PrestaShop\Adapter\ContainerBuilder).
+ * This factory is able to fetch the necessary parameters itself and builds the database connection for ModuleRepository.
+ *
+ * WARNING: this factory is only to be used in the specific cases mentioned above, for any other case please use the
+ * 'prestashop.module_kernel.repository' or 'prestashop.bundle.repository.module' depending on your needs.
+ */
+class ModuleRepositoryFactory
+{
+    /**
+     * @var self
+     */
+    private static $instance;
+
+    /**
+     * @var string
+     */
+    private $parameters;
+
+    /**
+     * @var string
+     */
+    private $environment;
+
+    /**
+     * @var ModuleRepository
+     */
+    private $moduleRepository;
+
+    /**
+     * @var string
+     */
+    private $parametersFile = '';
+
+    public static function getInstance()
+    {
+        if (!self::$instance) {
+            self::$instance = new ModuleRepositoryFactory();
+        }
+
+        return self::$instance;
+    }
+
+    /**
+     * @param string|null $environment
+     * @param array|null $parameters
+     */
+    public function __construct($environment = null, array $parameters = null)
+    {
+        $this->environment = $environment;
+        if (null === $environment) {
+            $this->environment = (new Environment())->getName();
+        }
+        $this->parameters = $parameters;
+    }
+
+    /**
+     * @return ModuleRepository
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function getRepository()
+    {
+        if (null !== $this->getParameters() && null === $this->moduleRepository) {
+            $databasePrefix = $this->parameters['database_prefix'];
```

Comment:
```suggestion
            $databasePrefix = $this->getParameters()['database_prefix'];
```

Reply:
`$this->parameters` is initialized in the getter so it works, but alright I will modify it

Reply:
The class is not final and we have an accessor,  this is why it's important to be consistent and use it everywhere.